### PR TITLE
task-856-963: add input (text) and select component examples

### PR DESCRIPTION
**Pull request by a121009**
The purpose of the PR is to add examples of our new Input (Text) and Select components as part of our QA and demonstration of functionality.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -52,8 +128,10 @@ document.addEventListener('DOMContentLoaded', () => {
...
   };
 
-  addItemButton.addEventListener('click', addItem);
-  subtractItemButton.addEventListener('click', subtractItem);
-  toggleModalButton.addEventListener('click', toggleModal);
   toggleThemeButton.addEventListener('click', toggleTheme);
-});
+};
+
+document.addEventListener('DOMContentLoaded', initModal);
```
Comment (a129204) [inline src/index.js]: could we create a single function which calls all of these different functions? Or is there a reason why that would be a bad idea?

Comment (a121009) [reply src/index.js]: We can. Originally I had everything happening within a single listener setup, and I pulled it apart to create a sort of separation of concerns. I don't think it is necessary here, but you've given me an idea. Stand by for incoming update!  :)

edit: I've wrapped all the init functions in an initializers object and then looped over them to add them on **DOMContentLoaded** event. What do you think?

File: src/index.js [hunk 2]
```diff
@@ -1,59 +1,135 @@
...
+  },
+};
 
-  addItemButton.addEventListener('click', addItem);
-  subtractItemButton.addEventListener('click', subtractItem);
-  toggleModalButton.addEventListener('click', toggleModal);
-  toggleThemeButton.addEventListener('click', toggleTheme);
-});
+for (const initializer in initializers) {
+  document.addEventListener('DOMContentLoaded', initializers[initializer]);
```
Comment (a129204) [inline src/index.js:134]: this is an interesting approach! This PR looks good to me!

**Review by a129204 — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (a129204) [reply src/index.js]: I think that's an interesting and good approach! I was thinking we could just call each of those functions in one function. But I like your approach!

**Pull request merged**